python/cudf/cudf/io/parquet.py

```diff
@@ -577,11 +577,26 @@ def read_parquet(
         )
     filepath_or_buffer = paths if paths else filepath_or_buffer
 
+    # Prepare remote-IO options
+    prefetch_options = kwargs.pop("prefetch_options", {})
+    if not ioutils._is_local_filesystem(fs):
+        method = prefetch_options.get("method", "parquet")
+        if method == "parquet":
+            prefetch_options = prefetch_options.update(
+                {
+                    "method": method,
+                    "columns": columns,
+                    # All paths must have the same row-group selection
+                    "row_groups": row_groups[0] if row_groups else None,
```

Comment:
Can we validate that all the row group entries are the same? Something like `if not all(r == row_groups[0] for r in row_groups): raise ValueError(...)`.

Reply:
Yeah, I actually tweaked the defaults a bit to make sure we don't try using the 'parquet' prefetcher if the user has asked for a strange multi-file row-group selection.